app/containers/filterContainer/filterSaveBox/titleBox.tsx

```diff
@@ -0,0 +1,271 @@
+import * as React from "react";
+import * as classNames from "classnames";
+import * as store from "store";
+import { withRouter, RouteComponentProps } from "react-router-dom";
+import { Picker } from "emoji-mart";
+import { withStyles } from "../../../helpers/withStylesHelper";
+import { BaseEmoji } from "emoji-mart/dist-es/utils/emoji-index/nimble-emoji-index";
+import { Filter } from "../../../api/member";
+import { ArticleSearchState } from "../../../components/articleSearch/records";
+import FilterSaveButton from "../../../components/filterContainer/filterSaveButton";
+import FilterResetButton from "../../../components/filterContainer/filterResetButton";
+import ScinapseInput from "../../../components/common/scinapseInput";
+import alertToast from "../../../helpers/makePlutoToastAction";
+import getQueryParamsObject from "../../../helpers/getQueryParamsObject";
+import Icon from "../../../icons";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import ClickAwayListener from "@material-ui/core/ClickAwayListener";
+import ActionTicketManager from "../../../helpers/actionTicketManager";
+const styles = require("./filterSaveBox.scss");
+
+interface TitleBoxProps {
+  hasFilterChanged: boolean;
+  isDropdownOpen: boolean;
+  onClickDropdownOpen: (value: React.SetStateAction<boolean>) => void;
+  onClickSaveBtn: (changedFilterReq: Filter | string) => void;
+  onClickFilterItem: (query: string, sort: Scinapse.ArticleSearch.SEARCH_SORT_OPTIONS, filter: Filter | null) => void;
+  articleSearchState: ArticleSearchState;
+}
+
+const FilterTitleBox: React.FunctionComponent<TitleBoxProps & RouteComponentProps<any>> = props => {
+  const {
+    hasFilterChanged,
+    onClickSaveBtn,
+    onClickFilterItem,
+    onClickDropdownOpen,
+    articleSearchState,
+    isDropdownOpen,
+  } = props;
+  const { savedFilterSet, searchInput, sort, isFilterSaveBoxLoading } = articleSearchState;
+
+  const [isOpenTitleInput, setIsOpenTitleInput] = React.useState(false);
+  const [isOpenEmojiPicker, setIsOpenEmojiPicker] = React.useState(false);
+  const [filterTitle, setFilterTitle] = React.useState("");
+
+  const rawQueryParamsObj = getQueryParamsObject(props.location.search);
+  const currentFilterStr = rawQueryParamsObj.filter;
+
+  function getSaveAndResetBtns() {
+    const currentSavedFilterStr = !!savedFilterSet ? savedFilterSet.filter : "";
+
+    return hasFilterChanged ? (
```

Comment:
for this long assignment, ternary operator is much difficult to look.
please use if-else statement.